src/azure-cli/azure/cli/command_modules/network/_template_builder.py

```diff
@@ -121,13 +120,8 @@ def _ag_subresource_id(_type, name):
         })
 
     frontend_ip_configs = []
-    if public_ip_id:
-        frontend_public_ip = _build_frontend_ip_config(cmd, frontend_public_ip_name,
-                                                       public_ip_id=public_ip_id,
-                                                       enable_private_link=enable_private_link,
-                                                       private_link_configuration_id=private_link_configuration_id)
-        frontend_ip_configs.append(frontend_public_ip)
-    if private_ip_address:
+
+    if private_ip_address is not None or public_ip_id is None:
```

Comment:
Can user provide both `private_ip_address` and `public_ip_id`?

Reply:
Sure, firsly, if user provide `--public-ip-address`, then it will go to line135-142 and if `--private-ip-address` is also provided,  then `if priavte_ip_address it not None` is True in line 124. 

The tests in test guid are all supported.